openupgradelib/openupgrade.py

```diff
@@ -834,6 +837,48 @@ def reactivate_workflow_transitions(cr, transition_conditions):
             (condition, transition_id))
 
 
+def log():
+    """Decorator for automatic logging of the executed method.
+    :param custom_logger: for defining an specific logger
+    :param details: if True, more details are given in the log"""
+    def wrap(func, custom_logger=None, details=False):
+        def wrapped_function(*args, **kwargs):
+            msg = "Executing method %s" % func.__name__
+            if details:
+                if args:
+                    msg += " with args %s" % str(args)
+                if kwargs:
+                    if args:
+                        msg += " and kwargs %s" % str(kwargs)
+                    else:
+                        msg += " with kwargs %s" % str(kwargs)
+            custom_logger.info(msg)
+            func(*args, **kwargs)
+        if custom_logger is None:
+            custom_logger = logger
+        return wrapped_function
+    return wrap
+
+
+def logged_progress(func, iterator, title="", args=None, kwargs=None):
```

Comment:
Yeah, but if you make:
```
foo(1, 2, 3, t=4, x=5)
```
you got:
```
TypeError: foo() got multiple values for keyword argument 't'
```
with the way I have put it, you should put:
```
foo(1, 2, t=4, args=[3], kwargs={'x': 5})
```